### PR TITLE
fixes to num change based on meeting

### PR DESCRIPTION
**Pull request by AlexDeLos**
Changes said in the meeting

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -269,7 +283,7 @@ def create_data_instance(graph, y_bus):
 
 # Create the data needed for the physics loss
 # return a torch_geometric.data.Data object for each instance
-def create_physics_data_instance(graph, y_bus):
+def create_physics_data_instance(graph, y_bus, missing, volt):
```
Comment (francescoopiccoli) [inline src/utils.py:286]: for readability we could add comments specifying the role and type of each parameters, its not super clear what are  `missing` and `volt` here, maybe a better naming

File: data_generation/subgraphs_methods.py [hunk 2]
```diff
@@ -108,13 +108,36 @@ def random_walk_neighbor_selection(full_net, starting_bus, subgraph_length):
...
+    old = []
+    new = []
+    #basis from all the variation in the network
+    #decided for a uniform distribution
+    ratio_increase = np.random.uniform(0.9,1.1)
     for i in range(len(test_net.load)):
-        test_net.load.at[i,'p_mw'] = np.abs(np.random.normal(test_net.load.at[i,'p_mw'], np.sqrt(abs(test_net.load.at[i,'p_mw']))))# goes from 0 - 500 (most under 100)
+        individual_variation = np.random.normal(0,0.05)
+        if test_net.load.at[i,'p_mw']==0:
+            test_net.load.at[i,'p_mw'] = test_net.load.at[i,'p_mw'] + 0.00000000001
```
Comment (francescoopiccoli) [inline data_generation/subgraphs_methods.py]:
```suggestion
            test_net.load.at[i,'p_mw'] =  0.00000000001
```

File: data_generation/subgraphs_methods.py [hunk 3]
```diff
@@ -108,13 +108,36 @@ def random_walk_neighbor_selection(full_net, starting_bus, subgraph_length):
...
-    for i in range(len(test_net.gen)):
-        test_net.gen.at[i,'p_mw'] = np.abs(np.random.normal(test_net.gen.at[i,'p_mw'], np.sqrt(abs(test_net.gen.at[i,'p_mw'])/100))) #goes from 0 - 1000
-        test_net.gen.at[i,'vm_pu'] = np.abs(np.random.normal(1, 1/100)) #goes from 0 - 1.1
+    #TODO
+    # We vary every value based on how big they are around their point.
+    # values should vary so that they are not too far from their original value.
+    # assume 80% correlation between every node.
+    # Power should increase as much as the load increases.
+    old = []
+    new = []
```
Comment (francescoopiccoli) [inline data_generation/subgraphs_methods.py]:
```suggestion
    old_p_mws= []
    new_p_mws = []
```

File: data_generation/subgraphs_methods.py [hunk 4]
```diff
@@ -108,13 +108,36 @@ def random_walk_neighbor_selection(full_net, starting_bus, subgraph_length):
...
+            test_net.load.at[i,'p_mw'] = test_net.load.at[i,'p_mw'] + 0.00000000001
+        old.append(test_net.load.at[i,'p_mw'])
+        test_net.load.at[i,'p_mw'] = test_net.load.at[i,'p_mw'] * (ratio_increase + individual_variation)
+        new.append(test_net.load.at[i,'p_mw'])
+        #same correlation of the P_mw
         test_net.load.at[i,'q_mvar'] =np.abs(np.random.normal(test_net.load.at[i,'q_mvar'], np.sqrt(abs(test_net.load.at[i,'q_mvar'])))) # goes from -50 to about 300
+    # Average value by which the load changes
+    div = np.divide(np.array(new),np.array(old))
+    # np.nan_to_num(div)
+    change = np.average(div)
```
Comment (francescoopiccoli) [inline data_generation/subgraphs_methods.py:135]: just for my understanding, each original `p_mw` of each load is multiplied by a number ranging from 0.85 and 1.15, then we take the average between all loads, which will be around 1.0, then we also multiply the `p_mw` of the generators by such value right? 
I don't remember exactly all the steps we discussed with Lucas but I remember we said something with increasing the value at one side and decreasing on the other, do I remember wrong?
no need to answer here, we can talk next meeting, just double check this step. 
Also maybe comments along some lines can help explain the rationale behind the code

File: data_generation/subgraphs_methods.py [hunk 5]
```diff
@@ -108,13 +108,36 @@ def random_walk_neighbor_selection(full_net, starting_bus, subgraph_length):
...
+
+    for i in range(len(test_net.gen)):
+        individual_variation = np.random.normal(0,0.05)
```
Comment (francescoopiccoli) [inline data_generation/subgraphs_methods.py]: we are not using the `individual_variation` here

File: data_generation/subgraphs_methods.py [hunk 6]
```diff
@@ -108,13 +108,36 @@ def random_walk_neighbor_selection(full_net, starting_bus, subgraph_length):
...
     test_net = pp.pandapowerNet(full_net.copy())
-    # We vary every value based on how big they are around their point
-    for i in range(len(test_net.gen)):
-        test_net.gen.at[i,'p_mw'] = np.abs(np.random.normal(test_net.gen.at[i,'p_mw'], np.sqrt(abs(test_net.gen.at[i,'p_mw'])/100))) #goes from 0 - 1000
-        test_net.gen.at[i,'vm_pu'] = np.abs(np.random.normal(1, 1/100)) #goes from 0 - 1.1
+    #TODO
+    # We vary every value based on how big they are around their point.
+    # values should vary so that they are not too far from their original value.
+    # assume 80% correlation between every node.
+    # Power should increase as much as the load increases.
```
Comment (francescoopiccoli) [inline data_generation/subgraphs_methods.py]: is this the TODO?

File: scratch.txt
Comment (francescoopiccoli) [inline scratch.txt:1]: do we use this txt file?

File: test.sh
Comment (francescoopiccoli) [inline test.sh:1]: maybe a comment at the top help explain what do we use this file for / how to use it

**Review by francescoopiccoli — APPROVED** [review]

File: data_generation/subgraphs_methods.py (hunk 6 above)
Comment (AlexDeLos) [reply data_generation/subgraphs_methods.py]: no

File: data_generation/subgraphs_methods.py (hunk 4 above)
Comment (AlexDeLos) [reply data_generation/subgraphs_methods.py:135]: added some extra comments to explain. But we want to increase the generators if the loads increase and decrease the generators if the loads decrease

File: scratch.txt
Comment (AlexDeLos) [reply scratch.txt:1]: as far as I can tell not on this merge. DELETED

File: src/utils.py (hunk 1 above)
Comment (AlexDeLos) [reply src/utils.py:286]: added comments

**Pull request merged**